### PR TITLE
bring juju/juju.py into life

### LINKED ISSUE #529: python-libjuju does not implement juju.get_controllers
https://github.com/juju/python-libjuju/blob/8a037110bede2bc40e95d7096ce1cc19c66c429b/juju/juju.py#L58-L63

### PR DESCRIPTION
**Pull request by cderici**
### Description

This PR adds some functions into the long dormant `Juju` class in `juju/juju.py`, cleans it up a little bit and add some tests for the added codes.

#### Includes 5 commits to:
- add `get_controllers` that returns a dictionary of info about the available controllers
- add `get_controller` that returns a ready-to-use `Controller` object that is connected to the controller with the given name
- remove a bunch of stub functions that were initially included for `budget` functionality (that is being deprecated)
- remove `get_agreements`
- fix function description

This also fixes #529 

### QA Steps

```bash
  $ tox -e integration -- tests/integration/test_juju.py
```

### Notes & Discussion

There might be more stub functions to be removed about functionalities that are being deprecated (I suspect `plans`). I simply don't know which ones are, so if you do, then please add a comment below and I'll update the PR.

### CODE REVIEW

File: juju/juju.py [hunk 1]
```diff
@@ -31,24 +40,12 @@ def autoload_credentials(self):
         """
         raise NotImplementedError()
 
-    def create_budget(self):
-        """Create a new budget.
-
-        """
-        raise NotImplementedError()
-
     def get_agreements(self):
```
Comment (SimonRichardson) [inline juju/juju.py:43]: Delete this one as well.

File: juju/juju.py [hunk 2]
```diff
@@ -95,15 +82,26 @@ def get_cloud(self, name):
...
         raise NotImplementedError()
 
-    def get_controller(self, name, include_passwords=False):
+    async def get_controller(self, name, include_passwords=False):
         """Get a controller by name.
 
         :param str name: Name of controller
         :param bool include_passwords: Include passwords for accounts
 
         (maybe move this to Cloud?)
```
Comment (SimonRichardson) [inline juju/juju.py:91]: Delete this line and mention that it will also try and connect.

**Comment by cderici** [comment]
[implement juju.py functions like juju.get_controllers](https://trello.com/c/m8BvbBi3/3952-implement-jujupy-functions-like-jujugetcontrollers)

**Review by SimonRichardson — APPROVED** [review]

**Comment by cderici** [comment]
$$merge$$

**Pull request merged**